### ISSUE TITLE
Format strings set in tabular editor lost after saving, closing and reopening the Power BI file

### ISSUE THREAD

**Issue by Rickmaurinus** [otykier/TabularEditor#702]
**Describe the bug**
Hi. In two of my projects using Tabular Editor I notice the following. I set the formatting string in tabular editor, save it and see the changes reflected in Power BI desktop. I then upload my file to the Power BI service, save the PBIX and close Power BI Desktop. The day after when I re-open my PBIX file, some of the measure's format strings (not all) have been removed. This is inconvenient as I have to re-apply them every time I work with the file. I have up to now done this at least 20 times, so it's not an incidental thing. 

Instead I tried to set the format string in PBI desktop again, however those options are greyed out and can't be set.

**Expected behavior**
I would expect the tool to hold on to the changed format string in Tabular Editor. I experience the issue with 
- Power BI desktop September, Oktober and November version of 2020 
- Tabular Editor version: [2.12.4]

The format strings I tried to set is: 
"#,##0;(#,##0)"
"#,##0.0%;(#,##0.0%)"

When coming back to the file after saving and reopening, is that the string has either been replaced by: 
"#,0"
"" (empty)

Any advise?

Best regards,
Rick

**Comment by otykier** [comment]
Closed as duplicate of #666

**Comment by Rickmaurinus** [comment]
> Closed as duplicate of #666

Hi @otykier ,

Please elaborate how this thread is a duplicate of #666 ? I think it is a different issue in the sense that the other thread loses its formatting when updating to a new power bi version. In this ticket I describe how the formatting is gone after saving the file, publishing the data, closing and then reopening the file, regardless of the Power BI updates. It also applies to PBIX files within the same version.

Best regards,
Rick de Groot

**Comment by otykier** [comment]
The same workaround applies (remove the Format annotation), and the underlying issue is the same: Power BI reapplies the settings specified in the annotation at various times, overwriting the settings specified in the FormatString from external tools. In any case, this is not a Tabular Editor issue - it should be fixed on the Power BI side.

**Comment by Rickmaurinus** [comment]
> The same workaround applies (remove the Format annotation), and the underlying issue is the same: Power BI reapplies the settings specified in the annotation at various times, overwriting the settings specified in the FormatString from external tools. In any case, this is not a Tabular Editor issue - it should be fixed on the Power BI side.

Hi Otykier. 

I appreciate your reponse. I've tried your workaround described at: https://github.com/otykier/TabularEditor/issues/666. 

I imagined it would work by: 
1. Changing the formatting string of measures in Tabular editor. 
2. Then running the script you provided (when the measures that were changed were selected). I did this in the 'Advanced scripting' tab. 
3. Save changes in Tabular editor. 

I've tried this on several of my files so far. However, when reopening the Power BI file at a later stage, the changes are lost every time. Am I doing something wrong? 

Best regards,
Rick de Groot
